### DISCUSSION TITLE
[Question] BottomSheet Size in Windows App

### DISCUSSION THREAD

**Discussion by buzbyh** [flet-dev/flet#1908] [Q&A] [answered]
### Question

Hello, I'm currently developing a Windows app using the BottomSheet control.
In version 0.4, the BottomSheet was created to match the width of the app, eliminating any left or right empty space.

However, after updating the 'flet' module to its latest version,
I've noticed that even when setting the internal container size to match the app's width (which is 1280),
it doesn't expand beyond a certain size, leaving empty space on the left and right.

Is there a way to make the BottomSheet cover the entire width without any empty space when using it?
Your insights and suggestions would be greatly appreciated.

### Code sample

```python
import flet as ft


def main(page: ft.Page):
    def bs_dismissed(e):
        print("Dismissed!")

    def show_bs(e):
        bs.open = True
        bs.update()

    def close_bs(e):
        bs.open = False
        bs.update()

    page.window_width = 1280
    page.window_height = 960
    
    bs = ft.BottomSheet(
        ft.Container(
            ft.Column(
                [
                    ft.Text("This is sheet's content!"),
                    ft.ElevatedButton("Close bottom sheet", on_click=close_bs),
                ],
                tight=True,
            ),
            padding=10,
            width=1280,
        ),
        open=True,
        on_dismiss=bs_dismissed,
    )
    page.overlay.append(bs)
    page.add(ft.ElevatedButton("Display bottom sheet", on_click=show_bs))

ft.app(target=main)
```

### ------------------------------------------------------

- [X] I have searched for answers to my question both in the issues and in previous discussions.

**Comment by plugeit** [comment]
you can create your custom bottom sheet using flet if it's not behaving as you intended it to be or go back to older version or copy specific files from older version and paste it in newer version related to bottom sheet best would be to make a custom control

**Comment by ndonkoHenri** [comment, marked as answer]
The solution is mentioned [here](https://api.flutter.dev/flutter/material/BottomSheet/constraints.html).
You have to set the `use_material3` theme's param to `False`. By so doing, the BS will automatically fill the page's width. (No more need to even specify a width.)
```py
def main(page: ft.Page):
  page.theme = ft.Theme(use_material3=False)
  ....
```

**Reply by buzbyh** [reply]
Thank you. I've confirmed that the method you provided works well for my needs.

Is there any alternative way to use_material3?
Personally, I find the design of use_material3 more appealing than use_material2, and I'm curious if there are any other options available.

**Reply by ndonkoHenri** [reply]
I don't think so. The flutter docs seem quite clear on this. 😕
[Here](https://m3.material.io/components/bottom-sheets/specs#3b256f38-c4fd-44e5-ba46-c5858b626ec7) are the material 3 specifications for the BottomSheet on which flutter based itself.

**Reply by buzbyh** [reply]
Thank you for your response.
